app/controllers/spree/admin/products_controller.rb

```diff
@@ -76,10 +76,13 @@ def bulk_update
 
         product_set.collection.each { |p| authorize! :update, p }
 
-        if product_set.save
+        saved = product_set.save
+        errors = product_and_variants_errors(product_set)
+
+        if saved && errors.empty?
```

Comment:
I think that you need to check only the errors. It doesn't matter if the product got saved but the variants had errors. And if the product wasn't saved, there should always be an error.
```suggestion
        product_set.save
        errors = product_and_variants_errors(product_set)

        if errors.empty?
```